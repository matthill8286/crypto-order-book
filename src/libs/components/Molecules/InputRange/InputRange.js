function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React, { useState } from 'react';
import { StyledInputRange, StyledInputRangeBubble, StyledInputRangeBubbleContainer, StyledInputRangeBubblePosition, StyledInputRangeBubbleTypo, StyledInputRangeContainer, StyledInputRangeMarkLine, StyledInputRangeMarks, StyledInputRangeMarkTypo, StyledInputRangeThumb } from "./InputRange.styled";
export var InputRange = function InputRange(props) {
  var _props$name = props.name,
      name = _props$name === void 0 ? 'input-range' : _props$name,
      _props$step = props.step,
      step = _props$step === void 0 ? '0.1' : _props$step,
      _props$markAmount = props.markAmount,
      markAmount = _props$markAmount === void 0 ? 0 : _props$markAmount,
      _props$min = props.min,
      min = _props$min === void 0 ? '0' : _props$min,
      _props$max = props.max,
      max = _props$max === void 0 ? '1' : _props$max,
      _props$value = props.value,
      value = _props$value === void 0 ? '1' : _props$value,
      _props$allowMoreThanM = props.allowMoreThanMax,
      allowMoreThanMax = _props$allowMoreThanM === void 0 ? false : _props$allowMoreThanM,
      _props$withBubble = props.withBubble,
      withBubble = _props$withBubble === void 0 ? true : _props$withBubble,
      _props$showMarks = props.showMarks,
      showMarks = _props$showMarks === void 0 ? false : _props$showMarks,
      onChange = props.onChange;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      isSliding = _useState2[0],
      setIsSliding = _useState2[1];

  var percentage = (+value - +min) * 100 / (+max - +min);

  var onChangeHandler = function onChangeHandler(ev) {
    if (onChange) {
      onChange(ev);
    }
  };

  var calcIntervalLabel = function calcIntervalLabel(interval) {
    return Math.round(interval * ((+max - +min) / markAmount) + +min);
  };

  return React.createElement(StyledInputRangeContainer, null, withBubble && React.createElement(StyledInputRangeBubbleContainer, null, React.createElement(StyledInputRangeBubblePosition, {
    style: {
      left: "".concat(percentage, "%")
    }
  }, React.createElement(StyledInputRangeBubble, {
    isSliding: isSliding
  }, React.createElement(StyledInputRangeBubbleTypo, {
    color: "primary",
    fontFamily: "default",
    limitLines: 1,
    tag: "span",
    weight: "regular"
  }, Math.round(parseFloat(value) * 10))))), React.createElement(StyledInputRangeThumb, {
    style: {
      left: "".concat(percentage, "%")
    }
  }), React.createElement(StyledInputRange, {
    type: "range",
    name: name,
    list: name,
    step: step,
    min: min,
    max: max,
    value: value,
    onMouseDown: function onMouseDown() {
      return setIsSliding(true);
    },
    onTouchStart: function onTouchStart() {
      return setIsSliding(true);
    },
    onMouseUp: function onMouseUp() {
      return setIsSliding(false);
    },
    onTouchEnd: function onTouchEnd() {
      return setIsSliding(false);
    },
    onChange: onChangeHandler
  }), showMarks && markAmount > 0 && React.createElement(StyledInputRangeMarks, null, Array.from({
    length: markAmount + 1
  }, function (_, interval) {
    return React.createElement(StyledInputRangeMarkLine, {
      key: interval,
      index: interval,
      markAmount: markAmount,
      left: (calcIntervalLabel(interval) - +min) * 100 / (+max - +min)
    }, React.createElement(StyledInputRangeMarkTypo, {
      weight: "semibold",
      fontSize: "lg",
      limitLines: 1
    }, calcIntervalLabel(interval), " ", allowMoreThanMax && interval === markAmount && '+'));
  })));
};
//# sourceMappingURL=InputRange.js.map