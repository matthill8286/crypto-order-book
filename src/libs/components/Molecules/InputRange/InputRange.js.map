{"version":3,"sources":["../../../../src/components/Molecules/InputRange/InputRange.tsx"],"names":["React","useState","StyledInputRange","StyledInputRangeBubble","StyledInputRangeBubbleContainer","StyledInputRangeBubblePosition","StyledInputRangeBubbleTypo","StyledInputRangeContainer","StyledInputRangeMarkLine","StyledInputRangeMarks","StyledInputRangeMarkTypo","StyledInputRangeThumb","InputRange","props","name","step","markAmount","min","max","value","allowMoreThanMax","withBubble","showMarks","onChange","isSliding","setIsSliding","percentage","onChangeHandler","ev","calcIntervalLabel","interval","Math","round","left","parseFloat","Array","from","length","_"],"mappings":";;;;;;;;;;;;AAAA,OAAOA,KAAP,IAAgBC,QAAhB,QAAgC,OAAhC;AACA,SACEC,gBADF,EAEEC,sBAFF,EAGEC,+BAHF,EAIEC,8BAJF,EAKEC,0BALF,EAMEC,yBANF,EAOEC,wBAPF,EAQEC,qBARF,EASEC,wBATF,EAUEC,qBAVF;AAwBA,OAAO,IAAMC,UAAqC,GAAG,SAAxCA,UAAwC,CAAAC,KAAK,EAAI;AAAA,oBAYxDA,KAZwD,CAE1DC,IAF0D;AAAA,MAE1DA,IAF0D,4BAEnD,aAFmD;AAAA,oBAYxDD,KAZwD,CAG1DE,IAH0D;AAAA,MAG1DA,IAH0D,4BAGnD,KAHmD;AAAA,0BAYxDF,KAZwD,CAI1DG,UAJ0D;AAAA,MAI1DA,UAJ0D,kCAI7C,CAJ6C;AAAA,mBAYxDH,KAZwD,CAK1DI,GAL0D;AAAA,MAK1DA,GAL0D,2BAKpD,GALoD;AAAA,mBAYxDJ,KAZwD,CAM1DK,GAN0D;AAAA,MAM1DA,GAN0D,2BAMpD,GANoD;AAAA,qBAYxDL,KAZwD,CAO1DM,KAP0D;AAAA,MAO1DA,KAP0D,6BAOlD,GAPkD;AAAA,8BAYxDN,KAZwD,CAQ1DO,gBAR0D;AAAA,MAQ1DA,gBAR0D,sCAQvC,KARuC;AAAA,0BAYxDP,KAZwD,CAS1DQ,UAT0D;AAAA,MAS1DA,UAT0D,kCAS7C,IAT6C;AAAA,yBAYxDR,KAZwD,CAU1DS,SAV0D;AAAA,MAU1DA,SAV0D,iCAU9C,KAV8C;AAAA,MAW1DC,QAX0D,GAYxDV,KAZwD,CAW1DU,QAX0D;;AAAA,kBAc1BtB,QAAQ,CAAC,KAAD,CAdkB;AAAA;AAAA,MAcrDuB,SAdqD;AAAA,MAc1CC,YAd0C;;AAe5D,MAAMC,UAAU,GAAI,CAAC,CAACP,KAAD,GAAS,CAACF,GAAX,IAAkB,GAAnB,IAA2B,CAACC,GAAD,GAAO,CAACD,GAAnC,CAAnB;;AAEA,MAAMU,eAAe,GAAG,SAAlBA,eAAkB,CAACC,EAAD,EAA6C;AACnE,QAAIL,QAAJ,EAAc;AACZA,MAAAA,QAAQ,CAACK,EAAD,CAAR;AACD;AACF,GAJD;;AAMA,MAAMC,iBAAiB,GAAG,SAApBA,iBAAoB,CAACC,QAAD;AAAA,WACxBC,IAAI,CAACC,KAAL,CAAWF,QAAQ,IAAI,CAAC,CAACZ,GAAD,GAAO,CAACD,GAAT,IAAgBD,UAApB,CAAR,GAA0C,CAACC,GAAtD,CADwB;AAAA,GAA1B;;AAGA,SACE,oBAAC,yBAAD,QACGI,UAAU,IACT,oBAAC,+BAAD,QACE,oBAAC,8BAAD;AAAgC,IAAA,KAAK,EAAE;AAAEY,MAAAA,IAAI,YAAKP,UAAL;AAAN;AAAvC,KACE,oBAAC,sBAAD;AAAwB,IAAA,SAAS,EAAEF;AAAnC,KACE,oBAAC,0BAAD;AACE,IAAA,KAAK,EAAC,SADR;AAEE,IAAA,UAAU,EAAC,SAFb;AAGE,IAAA,UAAU,EAAE,CAHd;AAIE,IAAA,GAAG,EAAC,MAJN;AAKE,IAAA,MAAM,EAAC;AALT,KAMGO,IAAI,CAACC,KAAL,CAAWE,UAAU,CAACf,KAAD,CAAV,GAA8B,EAAzC,CANH,CADF,CADF,CADF,CAFJ,EAiBE,oBAAC,qBAAD;AAAuB,IAAA,KAAK,EAAE;AAAEc,MAAAA,IAAI,YAAKP,UAAL;AAAN;AAA9B,IAjBF,EAkBE,oBAAC,gBAAD;AACE,IAAA,IAAI,EAAC,OADP;AAEE,IAAA,IAAI,EAAEZ,IAFR;AAGE,IAAA,IAAI,EAAEA,IAHR;AAIE,IAAA,IAAI,EAAEC,IAJR;AAKE,IAAA,GAAG,EAAEE,GALP;AAME,IAAA,GAAG,EAAEC,GANP;AAOE,IAAA,KAAK,EAAEC,KAPT;AAQE,IAAA,WAAW,EAAE;AAAA,aAAMM,YAAY,CAAC,IAAD,CAAlB;AAAA,KARf;AASE,IAAA,YAAY,EAAE;AAAA,aAAMA,YAAY,CAAC,IAAD,CAAlB;AAAA,KAThB;AAUE,IAAA,SAAS,EAAE;AAAA,aAAMA,YAAY,CAAC,KAAD,CAAlB;AAAA,KAVb;AAWE,IAAA,UAAU,EAAE;AAAA,aAAMA,YAAY,CAAC,KAAD,CAAlB;AAAA,KAXd;AAYE,IAAA,QAAQ,EAAEE;AAZZ,IAlBF,EAgCGL,SAAS,IAAIN,UAAU,GAAG,CAA1B,IACC,oBAAC,qBAAD,QACGmB,KAAK,CAACC,IAAN,CAAW;AAAEC,IAAAA,MAAM,EAAErB,UAAU,GAAG;AAAvB,GAAX,EAAuC,UAACsB,CAAD,EAAIR,QAAJ;AAAA,WACtC,oBAAC,wBAAD;AACE,MAAA,GAAG,EAAEA,QADP;AAEE,MAAA,KAAK,EAAEA,QAFT;AAGE,MAAA,UAAU,EAAEd,UAHd;AAIE,MAAA,IAAI,EAAG,CAACa,iBAAiB,CAACC,QAAD,CAAjB,GAA8B,CAACb,GAAhC,IAAuC,GAAxC,IAAgD,CAACC,GAAD,GAAO,CAACD,GAAxD;AAJR,OAKE,oBAAC,wBAAD;AAA0B,MAAA,MAAM,EAAC,UAAjC;AAA4C,MAAA,QAAQ,EAAC,IAArD;AAA0D,MAAA,UAAU,EAAE;AAAtE,OACGY,iBAAiB,CAACC,QAAD,CADpB,OACiCV,gBAAgB,IAAIU,QAAQ,KAAKd,UAAjC,IAA+C,GADhF,CALF,CADsC;AAAA,GAAvC,CADH,CAjCJ,CADF;AAkDD,CA5EM","sourcesContent":["import React, { useState } from 'react'\nimport {\n  StyledInputRange,\n  StyledInputRangeBubble,\n  StyledInputRangeBubbleContainer,\n  StyledInputRangeBubblePosition,\n  StyledInputRangeBubbleTypo,\n  StyledInputRangeContainer,\n  StyledInputRangeMarkLine,\n  StyledInputRangeMarks,\n  StyledInputRangeMarkTypo,\n  StyledInputRangeThumb,\n} from './InputRange.styled'\n\nexport type InputRangeProps = Pick<\n  JSX.IntrinsicElements['input'],\n  'name' | 'step' | 'min' | 'max' | 'value'\n> & {\n  markAmount?: number\n  allowMoreThanMax?: boolean\n  withBubble?: boolean\n  showMarks?: boolean\n  onChange?: React.ChangeEventHandler<HTMLInputElement>\n}\n\nexport const InputRange: React.FC<InputRangeProps> = props => {\n  const {\n    name = 'input-range',\n    step = '0.1',\n    markAmount = 0,\n    min = '0',\n    max = '1',\n    value = '1',\n    allowMoreThanMax = false,\n    withBubble = true,\n    showMarks = false,\n    onChange,\n  } = props\n\n  const [isSliding, setIsSliding] = useState(false)\n  const percentage = ((+value - +min) * 100) / (+max - +min)\n\n  const onChangeHandler = (ev: React.ChangeEvent<HTMLInputElement>) => {\n    if (onChange) {\n      onChange(ev)\n    }\n  }\n\n  const calcIntervalLabel = (interval: number): number =>\n    Math.round(interval * ((+max - +min) / markAmount) + +min)\n\n  return (\n    <StyledInputRangeContainer>\n      {withBubble && (\n        <StyledInputRangeBubbleContainer>\n          <StyledInputRangeBubblePosition style={{ left: `${percentage}%` }}>\n            <StyledInputRangeBubble isSliding={isSliding}>\n              <StyledInputRangeBubbleTypo\n                color=\"primary\"\n                fontFamily=\"default\"\n                limitLines={1}\n                tag=\"span\"\n                weight=\"regular\">\n                {Math.round(parseFloat(value as string) * 10)}\n              </StyledInputRangeBubbleTypo>\n            </StyledInputRangeBubble>\n          </StyledInputRangeBubblePosition>\n        </StyledInputRangeBubbleContainer>\n      )}\n      <StyledInputRangeThumb style={{ left: `${percentage}%` }} />\n      <StyledInputRange\n        type=\"range\"\n        name={name}\n        list={name}\n        step={step}\n        min={min}\n        max={max}\n        value={value}\n        onMouseDown={() => setIsSliding(true)}\n        onTouchStart={() => setIsSliding(true)}\n        onMouseUp={() => setIsSliding(false)}\n        onTouchEnd={() => setIsSliding(false)}\n        onChange={onChangeHandler}\n      />\n      {showMarks && markAmount > 0 && (\n        <StyledInputRangeMarks>\n          {Array.from({ length: markAmount + 1 }, (_, interval) => (\n            <StyledInputRangeMarkLine\n              key={interval}\n              index={interval}\n              markAmount={markAmount}\n              left={((calcIntervalLabel(interval) - +min) * 100) / (+max - +min)}>\n              <StyledInputRangeMarkTypo weight=\"semibold\" fontSize=\"lg\" limitLines={1}>\n                {calcIntervalLabel(interval)} {allowMoreThanMax && interval === markAmount && '+'}\n              </StyledInputRangeMarkTypo>\n            </StyledInputRangeMarkLine>\n          ))}\n        </StyledInputRangeMarks>\n      )}\n    </StyledInputRangeContainer>\n  )\n}\n"],"file":"InputRange.js"}