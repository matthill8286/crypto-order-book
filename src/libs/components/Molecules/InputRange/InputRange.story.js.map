{"version":3,"sources":["../../../../src/components/Molecules/InputRange/InputRange.story.tsx"],"names":["withState","boolean","number","text","storiesOf","React","InputRange","stories","module","add","value","store","onChange","event","target","set","knobs","min","max","step","markAmount","showMarks","withBubble","allowMoreThanMax","state"],"mappings":";;AAAA,SAASA,SAAT,QAA0B,0BAA1B;AACA,SAASC,OAAT,EAAkBC,MAAlB,EAAkCC,IAAlC,QAA8C,wBAA9C;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,UAAT;AAEA,IAAMC,OAAO,GAAGH,SAAS,CAAC,0CAAD,EAA6CI,MAA7C,CAAzB;AAEAD,OAAO,CAACE,GAAR,CACE,SADF,EAEET,SAAS,CAAC;AAAEU,EAAAA,KAAK,EAAE;AAAT,CAAD,EAAmB,UAAAC,KAAK,EAAI;AACnC,MAAMC,QAAQ,GAAG,SAAXA,QAAW,CAAAC,KAAK,EAAI;AAAA,QAEZH,KAFY,GAGpBG,KAHoB,CAEtBC,MAFsB,CAEZJ,KAFY;AAIxBC,IAAAA,KAAK,CAACI,GAAN,CAAU;AAAEL,MAAAA,KAAK,EAALA;AAAF,KAAV;AACD,GALD;;AAOA,MAAMM,KAAK,GAAG,SAARA,KAAQ;AAAA,WAAwB;AACpCC,MAAAA,GAAG,EAAEd,IAAI,CAAC,KAAD,EAAQ,GAAR,CAD2B;AAEpCe,MAAAA,GAAG,EAAEf,IAAI,CAAC,KAAD,EAAQ,GAAR,CAF2B;AAGpCgB,MAAAA,IAAI,EAAEhB,IAAI,CAAC,MAAD,EAAS,KAAT,CAH0B;AAIpCiB,MAAAA,UAAU,EAAElB,MAAM,CAAC,YAAD,EAAe,CAAf,CAJkB;AAKpCmB,MAAAA,SAAS,EAAEpB,OAAO,CAAC,YAAD,EAAe,IAAf,CALkB;AAMpCqB,MAAAA,UAAU,EAAErB,OAAO,CAAC,aAAD,EAAgB,IAAhB,CANiB;AAOpCsB,MAAAA,gBAAgB,EAAEtB,OAAO,CAAC,kBAAD,EAAqB,KAArB;AAPW,KAAxB;AAAA,GAAd;;AAUA,SAAO,oBAAC,UAAD;AAAY,IAAA,KAAK,EAAEU,KAAK,CAACa,KAAN,CAAYd,KAA/B;AAAsC,IAAA,QAAQ,EAAEE;AAAhD,KAA8DI,KAAK,EAAnE,EAAP;AACD,CAnBQ,CAFX","sourcesContent":["import { withState } from '@dump247/storybook-state'\nimport { boolean, number, select, text } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport React from 'react'\nimport { InputRange, InputRangeProps } from './InputRange'\n\nconst stories = storiesOf('Design System/Molecules/InputRange/Basic', module)\n\nstories.add(\n  'default',\n  withState({ value: '0.5' }, store => {\n    const onChange = event => {\n      const {\n        target: { value },\n      } = event\n      store.set({ value })\n    }\n\n    const knobs = (): InputRangeProps => ({\n      min: text('min', '0'),\n      max: text('max', '1'),\n      step: text('step', '0.1'),\n      markAmount: number('markAmount', 1),\n      showMarks: boolean('Show Marks', true),\n      withBubble: boolean('With Bubble', true),\n      allowMoreThanMax: boolean('allowMoreThanMax', false),\n    })\n\n    return <InputRange value={store.state.value} onChange={onChange} {...knobs()} />\n  })\n)\n"],"file":"InputRange.story.js"}