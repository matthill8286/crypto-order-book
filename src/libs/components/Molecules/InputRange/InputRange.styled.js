import { css, media, saiyanTheme, styled } from "../../../styles";
import { Typo } from "../../Atoms/Typography";

var issaiyan = function issaiyan(theme) {
  return theme.name === saiyanTheme.name;
};

export var StyledInputRangeContainer = styled.div.withConfig({
  displayName: "InputRangestyled__StyledInputRangeContainer",
  componentId: "sc-10ybtf3-0"
})(["padding:", ";flex-grow:1;", "{padding:", ";}"], function (_ref) {
  var theme = _ref.theme;
  return theme.spacing.base.md;
}, media.maxSm, function (_ref2) {
  var theme = _ref2.theme;
  return "".concat(theme.spacing.base.md, " ").concat(theme.spacing.base.lg);
});
export var StyledInputRangeBubbleContainer = styled.div.withConfig({
  displayName: "InputRangestyled__StyledInputRangeBubbleContainer",
  componentId: "sc-10ybtf3-1"
})(["height:6rem;"]);
export var Animation = function Animation(_ref3) {
  var theme = _ref3.theme;
  return css(["animation:wobble 2s ease-in-out 1 alternate;@keyframes wobble{0%{transform:scale(1);}20%{transform:scale(1.15);}30%{transform:scale(1);}40%{transform:scale(1.1);}50%{transform:scale(1);}60%{transform:scale(1.05);}70%{transform:scale(1);}80%{transform:scale(1.02);}100%{transform:scale(1);}}"]);
};
export var StyledInputRangeBubblePosition = styled.output.withConfig({
  displayName: "InputRangestyled__StyledInputRangeBubblePosition",
  componentId: "sc-10ybtf3-2"
})(["", ""], function (_ref4) {
  var theme = _ref4.theme;
  return css(["width:", ";height:", ";position:relative;display:flex;justify-content:center;align-items:center;transform:translateX(-50%) ", ";", "{transform:translateX(-50%) ", ";}"], issaiyan(theme) ? "4rem" : "3.5rem", issaiyan(theme) ? "4rem" : "3.5rem", !issaiyan(theme) && " rotate(45deg)", media.maxMd, issaiyan(theme) ? " translateY(".concat(theme.spacing.base.sm, ")") : " translateY(".concat(theme.spacing.base.xs, ") rotate(45deg)"));
});
export var StyledInputRangeBubble = styled.output.withConfig({
  displayName: "InputRangestyled__StyledInputRangeBubble",
  componentId: "sc-10ybtf3-3"
})(["", ""], function (_ref5) {
  var theme = _ref5.theme,
      isSliding = _ref5.isSliding;
  return css(["box-sizing:border-box;width:100%;height:100%;display:flex;justify-content:center;align-items:center;border:2px solid ", ";", " ", ""], theme.color.primary, issaiyan(theme) && "border-radius: 50%;", isSliding && Animation);
});
export var StyledInputRangeBubbleTypo = styled(Typo).withConfig({
  displayName: "InputRangestyled__StyledInputRangeBubbleTypo",
  componentId: "sc-10ybtf3-4"
})(["", ""], function (_ref6) {
  var theme = _ref6.theme;
  return css(["font-size:", ";transform:", ";line-height:", ";padding:0;", "{font-size:", ";}"], !issaiyan(theme) ? theme.font.size.md : theme.font.size.xl, !issaiyan(theme) ? " translateY(4px) rotate(-45deg)" : "translateY(0)", theme.font.lineHeight.xl, media.maxMd, !issaiyan(theme) ? theme.font.size.xs : theme.font.size.md);
});
export var StyledInputRangeThumb = styled.div.withConfig({
  displayName: "InputRangestyled__StyledInputRangeThumb",
  componentId: "sc-10ybtf3-5"
})(["-webkit-appearance:none;box-shadow:", ";height:1px;width:1px;background:", ";cursor:pointer;position:relative;&:after{content:'';position:absolute;height:", ";width:", ";margin-left:-", ";border-radius:50%;background:", ";border:", ";}"], function (_ref7) {
  var theme = _ref7.theme;
  return theme.dimension.elevationLevel1;
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.color.primary;
}, function (_ref9) {
  var theme = _ref9.theme;
  return theme.spacing.base.md;
}, function (_ref10) {
  var theme = _ref10.theme;
  return theme.spacing.base.md;
}, function (_ref11) {
  var theme = _ref11.theme;
  return theme.spacing.base.sm;
}, function (_ref12) {
  var theme = _ref12.theme;
  return theme.color.white;
}, function (_ref13) {
  var theme = _ref13.theme;
  return "5px solid ".concat(theme.color.primary);
});

var InputRangeThumbArea = function InputRangeThumbArea(_ref14) {
  var theme = _ref14.theme;
  return css(["::-webkit-slider-thumb{-webkit-appearance:none;height:1px;width:1px;border-radius:50%;border:2px solid ", ";transform:translateY(", ") scale(", ");", "{transform:translateY(", ") scale(", ");}opacity:0;cursor:pointer;}::-moz-range-thumb{-webkit-appearance:none;height:1px;width:1px;border:2px solid ", ";transform:scale(", ");border-radius:50%;opacity:0;border:none;cursor:pointer;}::-ms-thumb{-webkit-appearance:none;height:1px;width:1px;border:2px solid ", ";border-radius:50%;opacity:0;cursor:pointer;}"], theme.color.white, theme.spacing.base.xxs, theme.defaultSpacing * 4, media.maxMd, theme.spacing.base.xxs, theme.defaultSpacing * 2, theme.color.white, theme.defaultSpacing * 4, theme.color.white);
};

var StyledInputRangeTrack = function StyledInputRangeTrack(_ref15) {
  var theme = _ref15.theme;
  return css(["::-webkit-slider-runnable-track{width:100%;height:", ";cursor:pointer;background:", ";box-shadow:", ";border-radius:", ";}:focus::-webkit-slider-runnable-track{background:", ";}::-moz-range-track{width:100%;height:", ";cursor:pointer;background:", ";box-shadow:", ";border-radius:", ";}::-ms-track{width:100%;height:", "px;cursor:pointer;background:transparent;border-color:transparent;border-width:16px 0;color:transparent;}::-ms-fill-lower{background:", ";box-shadow:", ";border-radius:", ";}::-ms-fill-upper{background:", ";box-shadow:", ";border-radius:", ";}:focus::-ms-fill-lower{background:", ";}:focus::-ms-fill-upper{background:", ";}"], theme.spacing.base.xs, theme.color.grey2, theme.dimension.elevationLevel1, theme.dimension.borderRadius1, theme.color.grey2, theme.spacing.base.xs, theme.color.grey2, theme.dimension.elevationLevel1, theme.dimension.borderRadius1, theme.spacing.base.xs, theme.color.grey2, theme.dimension.elevationLevel1, theme.dimension.borderRadius1, theme.color.grey2, theme.dimension.elevationLevel1, theme.dimension.borderRadius1, theme.color.grey2, theme.color.grey2);
};

export var StyledInputRange = styled.input.withConfig({
  displayName: "InputRangestyled__StyledInputRange",
  componentId: "sc-10ybtf3-6"
})(["width:100%;height:", ";-webkit-appearance:none;:focus{outline:none;}", " ", ""], function (_ref16) {
  var theme = _ref16.theme;
  return theme.spacing.base.lg;
}, InputRangeThumbArea, StyledInputRangeTrack);
export var StyledInputRangeMarks = styled.div.withConfig({
  displayName: "InputRangestyled__StyledInputRangeMarks",
  componentId: "sc-10ybtf3-7"
})(["position:relative;padding-bottom:4rem;"]);
export var StyledInputRangeMarkLine = styled.div.withConfig({
  displayName: "InputRangestyled__StyledInputRangeMarkLine",
  componentId: "sc-10ybtf3-8"
})(["left:", "%;top:-4px;position:absolute;height:", ";width:1px;background-color:", ";", "{display:", ";}", "{display:", ";}"], function (_ref17) {
  var left = _ref17.left;
  return left;
}, function (_ref18) {
  var theme = _ref18.theme;
  return theme.spacing.base.md;
}, function (_ref19) {
  var theme = _ref19.theme;
  return theme.color.grey3;
}, media.maxMd, function (_ref20) {
  var index = _ref20.index,
      markAmount = _ref20.markAmount;
  var showFirstLastAndEverySecond = index === 0 || index === markAmount || index % 2 === 0;
  return showFirstLastAndEverySecond ? 'block' : 'none';
}, media.maxSm, function (_ref21) {
  var index = _ref21.index,
      markAmount = _ref21.markAmount;
  var showFirstLastAndEveryFifth = index === 0 || index === markAmount || index % 5 === 0;
  return showFirstLastAndEveryFifth ? 'block' : 'none';
});
export var StyledInputRangeMarkTypo = styled(Typo).withConfig({
  displayName: "InputRangestyled__StyledInputRangeMarkTypo",
  componentId: "sc-10ybtf3-9"
})(["position:absolute;max-width:unset;overflow:unset;text-overflow:ellipsis;top:", ";transform:translateX(-50%);"], function (_ref22) {
  var theme = _ref22.theme;
  return theme.spacing.base.md;
});
//# sourceMappingURL=InputRange.styled.js.map