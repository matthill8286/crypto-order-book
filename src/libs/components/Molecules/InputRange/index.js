export * from "./InputRange";
//# sourceMappingURL=index.js.map