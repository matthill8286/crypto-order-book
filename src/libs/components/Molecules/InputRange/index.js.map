{"version":3,"sources":["../../../../src/components/Molecules/InputRange/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './InputRange'\n"],"file":"index.js"}