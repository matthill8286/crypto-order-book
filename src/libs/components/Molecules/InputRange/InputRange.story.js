function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { withState } from '@dump247/storybook-state';
import { boolean, number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { InputRange } from "./InputRange";
var stories = storiesOf('Design System/Molecules/InputRange/Basic', module);
stories.add('default', withState({
  value: '0.5'
}, function (store) {
  var onChange = function onChange(event) {
    var value = event.target.value;
    store.set({
      value: value
    });
  };

  var knobs = function knobs() {
    return {
      min: text('min', '0'),
      max: text('max', '1'),
      step: text('step', '0.1'),
      markAmount: number('markAmount', 1),
      showMarks: boolean('Show Marks', true),
      withBubble: boolean('With Bubble', true),
      allowMoreThanMax: boolean('allowMoreThanMax', false)
    };
  };

  return React.createElement(InputRange, _extends({
    value: store.state.value,
    onChange: onChange
  }, knobs()));
}));
//# sourceMappingURL=InputRange.story.js.map