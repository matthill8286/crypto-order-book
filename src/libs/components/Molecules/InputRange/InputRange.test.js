import React from 'react';
import { mountWithTheme, renderWithTheme } from "../../../testRenderer";
import { InputRange } from "./InputRange";
describe('InputRange', function () {
  it('should render', function () {
    var wrapper = renderWithTheme(React.createElement(InputRange, null));
    expect(wrapper).toMatchSnapshot();
  });
  it('should fire change event', function () {
    var onChange = jest.fn();
    var wrapper = mountWithTheme(React.createElement(InputRange, {
      onChange: onChange
    }));
    var input = wrapper.find('input');
    input.simulate('change');
    expect(onChange).toBeCalledTimes(1);
  });
});
//# sourceMappingURL=InputRange.test.js.map