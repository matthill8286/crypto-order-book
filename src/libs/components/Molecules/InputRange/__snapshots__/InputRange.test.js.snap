// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRange should render 1`] = `
<div
  className="InputRangestyled__StyledInputRangeContainer-sc-10ybtf3-0 kUxsRe"
>
  <div
    className="InputRangestyled__StyledInputRangeBubbleContainer-sc-10ybtf3-1 jKSnVa"
  >
    <output
      className="InputRangestyled__StyledInputRangeBubblePosition-sc-10ybtf3-2 TVQYH"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      <output
        className="InputRangestyled__StyledInputRangeBubble-sc-10ybtf3-3 cWCVac"
      >
        <span
          className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 jtbRpZ InputRangestyled__StyledInputRangeBubbleTypo-sc-10ybtf3-4 cjDDdZ"
          color="primary"
          fontFamily="default"
        >
          10
        </span>
      </output>
    </output>
  </div>
  <div
    className="InputRangestyled__StyledInputRangeThumb-sc-10ybtf3-5 jnrEVz"
    style={
      Object {
        "left": "100%",
      }
    }
  />
  <input
    className="InputRangestyled__StyledInputRange-sc-10ybtf3-6 bhLsAS"
    list="input-range"
    max="1"
    min="0"
    name="input-range"
    onChange={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    step="0.1"
    type="range"
    value="1"
  />
</div>
`;
